### PR TITLE
Adding the SqlClient Stress Test framework.

### PR DESCRIPTION
**Pull request by saurabh500**
This PR contains the StressTest framework for ADO.Net providers to build their stress tests against. 
Currently this is intended to be used by SqlClient stress tests. 

The Framework reads the configs and sets up the provider specific objects and allows execution of the stress tests.

### CODE REVIEW

**Comment by saurabh500** [comment]
cc @corivera @geleems

**Comment by saurabh500** [comment]
@dotnet-bot Test Innerloop Windows_NT Debug Build and Test please

**Comment by saurabh500** [comment]
@dotnet-bot Test Innerloop CentOS7.1 Release Build and Test please
@dotnet-bot Test Innerloop Windows_NT Release Build and Test please

**Comment by saurabh500** [comment]
Test Innerloop Windows_NT Release Build and Test please

File: src/System.Data.SqlClient/tests/StressTests/System.Data.StressFramework/DataSource.cs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        {
+            this.Name = name;
+            this.Type = type;
+            this.IsDefault = isDefault;
+        }
+
+        /// <summary>
+        /// this method is used to create the data source, based on its type
+        /// </summary>
+        static public DataSource Create(string name, DataSourceType sourceType, bool isDefault, IDictionary<string, string> properties)
```
Comment (corivera) [inline src/System.Data.SqlClient/tests/StressTests/System.Data.StressFramework/DataSource.cs]: public static

File: src/System.Data.SqlClient/tests/StressTests/System.Data.StressFramework/DataStressReader.cs [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+        {
+            InitSqlTypes();
+            InitClrTypes();
+        }
+
+        private static void InitSqlTypes()
+        {
+            s_sqlTypes = new Dictionary<Type, Func<DataStressReader, int, CancellationToken, Random, Task<object>>>();
+
+            // TODO: It is necessary to unwrap the ".Value" of the returned object?
```
Comment (corivera) [inline src/System.Data.SqlClient/tests/StressTests/System.Data.StressFramework/DataStressReader.cs]: Do we need this TODO for anything?

Comment (saurabh500) [inline src/System.Data.SqlClient/tests/StressTests/System.Data.StressFramework/DataStressReader.cs]: No. I should clean this up. Thanks for pointing

File: src/System.Data.SqlClient/tests/StressTests/System.Data.StressFramework/StressConfigReader.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+using System.Xml;
+using System.Xml.XPath;
+using static Stress.Data.DataStressSettings;
+
+namespace Stress.Data
+{
+    /// <summary>
+    /// Reads the configuration from a configuration file and provides the configuration
+    /// </summary>
+    class StressConfigReader
```
Comment (corivera) [inline src/System.Data.SqlClient/tests/StressTests/System.Data.StressFramework/StressConfigReader.cs]: Needs a visibility modifier

File: src/System.Data.SqlClient/tests/StressTests/System.Data.StressFramework/DataStressFactory.cs [hunk 4]
```diff
@@ -0,0 +1,956 @@
...
+
+        public void Dispose()
+        {
+            GC.SuppressFinalize(this);
+        }
+
+        public abstract string GetParameterName(string pName);
+
+
+        // TODO: can use use AutoIncrement on schema column infor instead?
```
Comment (corivera) [inline src/System.Data.SqlClient/tests/StressTests/System.Data.StressFramework/DataStressFactory.cs]: Unnecessary TODO?

**Comment by corivera** [comment]
LGTM after comments.

**Comment by saurabh500** [comment]
Test Innerloop Windows_NT Release Build and Test please

**Comment by saurabh500** [comment]
Test Innerloop Windows_NT Debug Build and Test please

**Comment by saurabh500** [comment]
Test Innerloop CentOS7.1 Release Build and Test please

**Pull request merged**